src/bsoncxx/json.cpp

```diff
@@ -274,15 +274,16 @@ std::string to_json(types::value value) {
 
     json_visitor v(ss, false, 0);
 
-    switch ((int)value.type()) {
-#define BSONCXX_ENUM(name, val)            \
-    case val:                              \
-        v.visit_value(value.get_##name()); \
-        break;
+    if(value.type()) {
```

Comment:
I don't think we can, but since value.type() is an optional, it seems safer to check before taking *(value.type()). Do you think it was the correct choice to skip the value if it's invalid?